vllm/engine/metrics.py

```diff
@@ -59,6 +59,10 @@ def __init__(self, labelnames: List[str], max_model_len: int):
             name="vllm:cpu_cache_usage_perc",
             documentation="CPU KV-cache usage. 1 means 100 percent usage.",
             labelnames=labelnames)
+        self.gauge_num_cumulative_preemption = Gauge(
```

Comment:
This could be a counter since it is monotonically increasing. Logging as a gauge is not best practices for Prometheus

Reply:
No, its not expensive because on the vllm side (which has the prometheus client), we are just incrementing a counter internally. The prometheus server then queries `/metrics` at whatever interval is specified by the `prometheus.yml` config (usually every 15s is pretty standard).

So I haven't measured really any impact